pyomo/core/tests/unit/test_numeric_expr.py

```diff
@@ -48,92 +49,64 @@
 from pyomo.core.base.var import _GeneralVarData
 
 from pyomo.repn import generate_standard_repn
+from pyomo.core.expr.numvalue import NumericValue
+
+
+class decompose_linear_term_wrapper(object):
```

Comment:
These functions currently use the `value` function. We should replace the use of `value` in these functions to utilize a function that can robustly compare expressions. I will create an issue for this.